distributed/executor.py

```diff
@@ -1798,3 +1829,62 @@ def futures_of(o, executor=None):
             raise CancelledError(bad)
 
     return list(futures)
+
+
+@contextmanager
+def serialize_for_publish(x):
```

Comment:
I recommend that you start with just a single collection:

    e.publish_dataset('name', my_collection)

You can get the keys from any dask collection, I think, by using the `._keys()` method.  There is also the `distributed.executor.py::futures_of` function, which may be useful.

My approach to making a serializable version would be to make a copy of the collection object and then swap out the dask attributes.

```python
from copy import copy
my_collection = copy(my_collection)
my_collection.dask = {}
```

However this won't work on Delayed values, which hold their dask graphs elsewhere.  Maybe we need a generic `._metadata()` method for dask objects or something.  Pinging @jcrist for whenever he gets back to civilization.